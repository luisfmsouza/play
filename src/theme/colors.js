export const skyLighter = "#F9FAFB";
export const skyLight = "#F4F6F8";
export const sky = "#DFE3E8";
export const skyDark = "#C4CDD5";
export const inkLightest = "#919EAB";
export const inkLighter = "#637381";
export const inkLight = "#454F5B";
export const ink = "#212B36";
