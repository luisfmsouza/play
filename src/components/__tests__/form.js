import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { ThemeProvider } from "styled-components";

import * as colors from "../../theme/colors";

import Form from "../form";

describe("Form component", () => {
  const build = state => {
    const store = {
      getState: () => state,
      dispatch: jest.fn(),
      subscribe: jest.fn()
    };

    return renderer
      .create(
        <Provider store={store}>
          <ThemeProvider theme={{ colors }}>
            <Form store={store} />
          </ThemeProvider>
        </Provider>
      )
      .toJSON();
  };

  describe("when there is playlists", () => {
    it("renders default component", () => {
      const state = {
        playlists: [[]],
        currentPlaylist: 0,
        currentVideo: null,
        form: { artist: "artist", title: "title", url: "https://url.com" },
        showForm: true
      };

      const expected = build(state);

      expect(expected).toMatchSnapshot();
    });
  });

  describe("when does not have playlists", () => {
    it("does not render the Form", () => {
      const state = {
        playlists: null,
        currentPlaylist: null,
        currentVideo: null,
        form: { artist: "", title: "", url: "" },
        showForm: false
      };

      const expected = build(state);

      expect(expected).toMatchSnapshot();
    });
  });
});
