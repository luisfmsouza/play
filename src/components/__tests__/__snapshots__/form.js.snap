// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component when does not have playlists does not render the Form 1`] = `null`;

exports[`Form component when there is playlists renders default component 1`] = `
<div
  className="sc-bdVaJa fKgYJA"
>
  <form
    onSubmit={[Function]}
  >
    <label
      className="sc-bwzfXH dnbGwd"
      htmlFor="artist"
    >
      Artist:
    </label>
    <input
      className="sc-htpNat gBBCDj"
      id="artist"
      onChange={[Function]}
      required={true}
      type="text"
      value="artist"
    />
    <label
      className="sc-bwzfXH dnbGwd"
      htmlFor="title"
    >
      Title:
    </label>
    <input
      className="sc-htpNat gBBCDj"
      id="title"
      onChange={[Function]}
      required={true}
      type="text"
      value="title"
    />
    <label
      className="sc-bwzfXH dnbGwd"
      htmlFor="videoId"
    >
      Youtube video ID:
    </label>
    <input
      className="sc-htpNat gBBCDj"
      id="videoId"
      onChange={[Function]}
      required={true}
      type="text"
      value="PtqAzo6ElhY"
    />
    <button
      className="sc-bxivhb dJVtKI"
      type="submit"
    >
      Add video
    </button>
  </form>
</div>
`;
